boards/arm/sensortile_box_pro/board.c

```diff
@@ -0,0 +1,63 @@
+/*
+ * Copyright (c) 2023 STMicroelectronics
+ *
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+#include <zephyr/kernel.h>
+#include <zephyr/init.h>
+#include <zephyr/drivers/gpio.h>
+#include <zephyr/sys/printk.h>
+#include <zephyr/usb/usb_device.h>
+#include <zephyr/drivers/uart.h>
+
+/*
+ * Use SPI2 as IMU sensors bus
+ */
+static int sensortile_box_pro_sensor_bus_init(void)
+{
+	const struct gpio_dt_spec mcu_sel =
+		GPIO_DT_SPEC_GET(DT_NODELABEL(sens_bus_mux), mcu_sel_gpios);
+
+	/* Switch sensor bus to SPI2 (MCU_SEL = 0) */
+	if (!gpio_is_ready_dt(&mcu_sel)) {
+		printk("MCU_SEL gpio port is not ready!\n");
+		return -ENODEV;
+	}
+
+	gpio_pin_configure_dt(&mcu_sel, GPIO_OUTPUT_LOW);
+	gpio_pin_set_dt(&mcu_sel, 0);
+
+	return 0;
+}
+
+/* needs to be done after GPIO driver init */
+SYS_INIT(sensortile_box_pro_sensor_bus_init, POST_KERNEL,
+	 CONFIG_KERNEL_INIT_PRIORITY_DEVICE);
+
+#ifdef CONFIG_USB_DEVICE_STACK
+/*
+ * Enable console on USB CDC_ACM
+ */
+static int sensortile_box_pro_usb_console_init(void)
+{
+	const struct device *const dev = DEVICE_DT_GET(DT_CHOSEN(zephyr_console));
+	uint32_t dtr = 0;
+
+	if (!device_is_ready(dev) || usb_enable(NULL)) {
+		return -1;
+	}
+
+	/* Poll if the DTR flag was set, optional */
+	while (!dtr) {
+		uart_line_ctrl_get(dev, UART_LINE_CTRL_DTR, &dtr);
+		k_sleep(K_MSEC(100));
+	}
```

Comment:
I think this belongs to the application, not the board file, @jfischer-no can you confirm?

Reply:
I confirm.

Reply:
> I think this belongs to the application, not the board file, @jfischer-no can you confirm?

This change come from the fact that I would like to have console on CDC_ACM by default, and be able to run a standard application, like hello world, w/o chnging it. Is this scenario possible? Am I missing something?

Reply:
yeah that's being work on in https://github.com/zephyrproject-rtos/zephyr/pull/64720

Reply:
ok, done!